.github/workflows/benchmark-pipeline.yaml

```
@@ -12,6 +12,9 @@ on:
       version:
         description: Version of project to be tested e.g. 0.37.0
         required: true
+      benchmark_path:
+        description: Path to the benchmark action
+        required: true
```

Makes sense for the input to be required but I think this will fail until we add the benchmark in the other ticket?

I think that's fine since we have the dependency between tickets but just wanted to mention it.

---

Yes, I thought about it too and forgot to mention in the PR notes. 

I thought the field would be required but right now the job would obviously fail since a part of the logic is temporarily missing. Maybe I can add a comment? Or is it okay to keep track of this here?

---

> Maybe I can add a comment?

+1 I would recommend changing this to false and adding a comment 👍 

```suggestion
        required: false # TODO: change to `true` when `"benchmark_path"` is specified in `projects/projects.json`
```

---

@vallss @nikimanoledaki I'm sorry I missed the CI error after committing your suggestion Niki :( I was hoping to get this merged before a meeting but its Fri 13th and I should have been more cautious! 👻 🤦‍♂️ 

I was wondering if we could add an `if` condition for the input being empty but AIUI `uses` is a special case and can't evaluate an input. https://github.com/orgs/community/discussions/45342

I think it may be simpler to add an empty workflow for Falco and reference it via the filename in which case we don't need to provide a ref like `@main`

> ./.github/workflows/{filename} for reusable workflows in the same repository.

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_iduses

I think with our naming the filename would be `./.github/workflows/falco_benchmark.yml` @nikimanoledaki is that correct?

@vallss Do you have time to look at this?

Don't worry if not! I can also take a look at it next week.

---

@rossf7 @nikimanoledaki yes, no problem! I can take a look at it starting from Monday (I'm on a short holiday right now). 

This is related to this error, right?
```
Invalid workflow file: .github/workflows/benchmark-pipeline.yaml#L66
invalid value workflow reference: no version specified
```
Would this be correct when we set a path value?

---

Thank you @vallss and yes this can definitely wait till you're back from your holiday!

> Would this be correct when we set a path value?

Yes, my reading of the [uses docs](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_iduses) is that if we specify a file path in the repo we don't need to specify an `@{ref}` and it will use the current commit.

However, I haven't done this before with GitHub Actions with a dynamic value for the workflow. So if you're able to try it out that would be great.

Likewise if you think there is a benefit to specifying an `@{ref}` we can use that approach instead.